packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -2119,6 +2182,56 @@ class _HorizontalInnerDimensionState extends ScrollableState {
     super.didChangeDependencies();
   }
 
+  // Returns the Future from calling ensureVisible for the ScrollPosition, as
+  // as well as the vertical ScrollableState instance so its context can be
+  // used to check for other ancestor Scrollables in executing ensureVisible.
+  @override
+  (List<Future<void>>, ScrollableState) _performEnsureVisible(
```

Comment:
> Is there a reason this inner state has to handle both the ensureVisible call for both axes?

Originally it was because the logic that bubbles up through scrollables/viewports in Scrollable.ensureVisible did not work for 2D since there are 2 scrollables and only one viewport. 

@goderbauer made the case that someone might want to call ensureVisible on only one of the two axes, but now I am realizing I did not quite add support for that like I intended, let me fix that.

Reply:
Oh wait I walked this through and I I think it is right. When calling the static method Scrollable.ensureVisible, the inner horizontal scrollable will always come up first, so it always passes through here.

Reply:
> Originally it was because the logic that bubbles up through scrollables/viewports in Scrollable.ensureVisible did not work for 2D since there are 2 scrollables and only one viewport.

Is that still the case? I was hoping we don't have to add that new axisDirection parameter to the interface since each scrollable knows its own axis.

Reply:
I think I added the Axis parameter to the Scrollable.ensureVisible method for 2 reasons. It is a static method so the returned scrollable's axis is unknown. The other reason was @goderbauer's suggestion that we permit users to call ensureVisible on individual axes of the 2DScrollable in order to allow different alignments in each axis. So I added Axis in order for folks to pick one specifically.

Thinking about this though, the current implementation of Scrollable.ensureVisible works on nested scrollables and applies the same axis for each scrollable as it bubbles up through them throughout ensureVisible. We don't allow folks to set a different alignment for each one now, so I don't think we need to here now. Developers can always get that level granularity and control by accessing the actual scroll position and calling ensureVisible there. 👍 

I removed Axis to keep the API cleaner.

Ultimately the axis does get passed to getOffsetToReveal later, since it is calling on that method on the viewport, which needs to know which axis' offset we are calculating.